packages/vscode-extension/src/network/providers/NetworkProvider.tsx

```diff
@@ -26,6 +27,59 @@ interface NetworkProviderProps extends NetworkTracker {
   toggleTimelineVisible: () => void;
   fetchAndOpenResponseInEditor: (networkLog: NetworkLog) => Promise<void>;
   getResponseBody: (networkLog: NetworkLog) => Promise<ResponseBodyData | undefined>;
+  getThemeData: (themeDescriptor: ThemeDescriptor) => Promise<ThemeData>;
+}
+
+function responseBodyListener(
+  resolve: (value: ResponseBodyData | undefined) => void,
```

Comment:
Passing the `resolve` callback here feels awkward.
I feel it'd make more sense for this function to return the Promise which resolves when the event is received.
```ts
function getResponseBody(messageId, requestId, responseBodiesRef): Promise<ResponseBodyData | undefined>
```

Reply:
Same goes for `themeListener`